src/helpers.php

```diff
@@ -317,4 +317,25 @@ function backpack_pro()
 
         return \PackageVersions\Versions::getVersion('backpack/pro');
     }
+
+    if (! function_exists('array_merge_recursive_distinct')) {
+        /**
+         * @param  array<int|string, mixed>  $array1
+         * @param  array<int|string, mixed>  $array2
+         * @return array<int|string, mixed>
+         */
+        function array_merge_recursive_distinct(array &$array1, array &$array2): array
+        {
+            $merged = $array1;
+            foreach ($array2 as $key => &$value) {
+                if (is_array($value) && isset($merged[$key]) && is_array($merged[$key])) {
+                    $merged[$key] = array_merge_recursive_distinct($merged[$key], $value);
+                } else {
+                    $merged[$key] = $value;
+                }
+            }
+
+            return $merged;
+        }
+    }
```

Comment:
I don't see this helper used anywhere in this PR. Or in CRUD. Is it something that the media package needs? In that case... it should probably live there.

Unless we eventually move that bit of code back to CRUD. In that case... yeah I guess it's ok to love here, but it's so messy right now and it prevents me from merging this PR.